lib/rack/mock.rb

```diff
@@ -210,9 +210,10 @@ def body
       #     ...
       #     res.body.should == "foo!"
       #   end
-      buffer = String.new
+      enum = super.each
```

Comment:
This is not guaranteed to work. The only way you could do this is by initializing the buffer during `each`:

``` ruby
body.each do |chunk|
  if buffer.nil?
    buffer = chunk.dup
  else
    buffer << chunk
  end
end
```
```

Reply:
@ioquatix -- Thanks for your swiftness!
Before I change it, could you point in which case it fail?